### PR TITLE
GHC-9.0 support for hls-tactics-plugin

### PR DESCRIPTION
**Pull request by isovector**
This is a fork of @anka-213's #2202 that makes all the tactics tests pass! :tada: 

The only issue is what to do about 9.2 support, which is broken by the compatibility layer here.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2581"></a>

### CODE REVIEW

**Comment by isovector** [comment]
OK, I think everything is good to go --- but having some issues getting Compat.Core to properly export TcType. Would someone more familiar with this scary module mind lending a hand?

**Review by pepeiborra — APPROVED** [review]
Awesome, thanks @isovector

File: test/functional/FunctionalCodeAction.hs [hunk 1]
```diff
@@ -353,6 +355,12 @@ unusedTermTests = testGroup "unused term code actions" [
               $ Just CodeActionQuickFix `notElem` kinds
     ]
 
+expectFailIfGhc92 :: String -> TestTree -> TestTree
+expectFailIfGhc92 reason =
```
Comment (jneira) [inline test/functional/FunctionalCodeAction.hs:359]: please, use the existing `knownBrokenForGhcVersions` like
https://github.com/haskell/haskell-language-server/blob/5714207bfe44b13b02726334f9caf883249e68a9/plugins/hls-hlint-plugin/test/Main.hs#L347

Comment (jneira) [reply test/functional/FunctionalCodeAction.hs:359]: from https://github.com/haskell/haskell-language-server/blob/master/hls-test-utils/src/Test/Hls/Util.hs#L32

**Pull request merged**